### ISSUE TITLE
Let's Encrypt renewals and  new certificate issues not working on 6.16

### ISSUE THREAD

**Issue by erpsolns** [virtualmin/virtualmin-gpl#283]
Let's Encrypt renewals nor new issues not working on 6.16

Requesting a certificate for ........com, www.......com, mail.......com from Let's Encrypt ..
.. request failed : Web-based validation failed : Failed to request certificate :
Traceback (most recent call last):
  File "/usr/libexec/webmin/webmin/acme_tiny.py", line 198, in <module>
    main(sys.argv[1:])
  File "/usr/libexec/webmin/webmin/acme_tiny.py", line 194, in main
    signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca, disable_check=args.disable_check, directory_url=args.directory_url, contact=args.contact)
  File "/usr/libexec/webmin/webmin/acme_tiny.py", line 149, in get_crt
    raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
ValueError: Challenge did not pass for mail.xxxxx.com: {u'status': u'invalid', u'challenges': [{u'status': u'invalid', u'url': u'https://acme-v02.api.letsencrypt.org/acme/chall-v3/12676405434/QysGUA', u'token': u'AaZ3Ltj2swZnPQO-FTOm4VE7nwPrj_VcBZy76dPiBp0', u'error': {u'status': 400, u'type': u'urn:ietf:params:acme:error:dns', u'detail': u"DNS problem: SERVFAIL looking up A for mail.pariharas.com - the domain's nameservers may be malfunctioning"}, u'validated': u'2021-04-28T02:52:02Z', u'type': u'http-01'}], u'identifier': {u'type': u'dns', u'value': u'mail............com'}, u'expires': u'2021-05-05T02:51:53Z'}
   DNS-based validation failed : Only the offical Let's Encrypt client supports DNS-based validation

**Comment by erpsolns** [comment]
This is another virtual server with only one domain Requesting a certificate forxxxxx.com, www.xxxxxx.com from Let's Encrypt ..
.. request failed : Web-based validation failed : Failed to request certificate :
Traceback (most recent call last):
  File "/usr/libexec/webmin/webmin/acme_tiny.py", line 198, in <module>
    main(sys.argv[1:])
  File "/usr/libexec/webmin/webmin/acme_tiny.py", line 194, in main
    signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca, disable_check=args.disable_check, directory_url=args.directory_url, contact=args.contact)
  File "/usr/libexec/webmin/webmin/acme_tiny.py", line 149, in get_crt
    raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
ValueError: Challenge did not pass for saiversity.com: {u'status': u'invalid', u'challenges': [{u'status': u'invalid', u'validationRecord': [{u'url': u'http://xxxxxx.com/.well-known/acme-challenge/7DV3kI_oczIkOLb5JFfP9HIQ_b1vIdcReEXN0jQbxQw', u'hostname': u'saiversity.com', u'addressUsed': u'91.121.65.81', u'port': u'80', u'addressesResolved': [u'91.121.65.81']}, {u'url': u'https://saiversity.com/2887-2/', u'hostname': u'saiversity.com', u'addressUsed': u'91.121.65.81', u'port': u'443', u'addressesResolved': [u'xx.xxx.xx.xx']}], u'url': u'https://acme-v02.api.letsencrypt.org/acme/chall-v3/12676358409/oYaFUg', u'token': u'7DV3kI_oczIkOLb5JFfP9HIQ_b1vIdcReEXN0jQbxQw', u'error': {u'status': 403, u'type': u'urn:ietf:params:acme:error:unauthorized', u'detail': u'Invalid response from https://xxxxxx.com/2887-2/ [xx.xxx.xx.xx]: 500'}, u'validated': u'2021-04-28T03:04:41Z', u'type': u'http-01'}], u'identifier': {u'type': u'dns', u'value': u'saiversity.com'}, u'expires': u'2021-05-05T02:49:20Z'}
   DNS-based validation failed : Only the offical Let's Encrypt client supports DNS-based validation

**Comment by chris001** [comment]
`DNS-based validation failed : Only the offical Let's Encrypt client supports DNS-based validation`
This is clearly the cause of the error.  Only the official Let's Encrypt client is allowed to do DNS-based validation of the cert renewal...

**Comment by swelljoe** [comment]
The DNS error is later. The validation server cannot retrieve the validation file from your server. You need to troubleshoot why the validation server can't request that file on your server. You probably have some proxy or redirect rules that don't allow .well-known directory to be left alone. You need to fix that.

DNS validation only happens for regular (non-wildcard) certs when web validation fails. And, it can't happen with the built-in acme-tiny client, because it does not support DNS validation. But if you fix your web validation you won't need it. DNS validation is not what you need to fix here, though it never hurts to install the standard ACME client (certbot), if you can. Old operating systems (e.g. CentOS 6) don't have it and don't have new enough versions of Python and libs to run it..

In short: Just fix your web redirects or proxy rules to allow everything in `.well-known` to be served normally.

**Comment by swelljoe** [comment]
It may also be that you don't have DNS records for all of the domains you have request a certificate for. The errors from acme-tiny are terrible, so I can't make a lot of sense out of it.

**Comment by erpsolns** [comment]
When mail.xxxxxxx.com is added with xxxxxxx.com andcord www.xxxxxxx.com the
letsencrypt denies certificate on almost all domains for us. Although mx
and A and C records are resolving to the host oroperly. This could be
because letsencrypt might treat mail servers differently

**Comment by iliarostovtsev** [comment]
Let's Encrypt doesn't care what's behind your FQDN. It needs to run validation, either web-based (by reading a challenge file from `.well-known` directory) or DNS, if `certbot`  package is installed.

Your `mail.` domain must be hosted on the same system where you run validation, and http to https redirects must be off (in case of web-based validation, for directory with challenge files at least).

Debugging is simple - simply put the small text file to your `mail.` domain and see if you can open it from your browser. If not, check web-server logs.
